feat: Add BlockButtons

lux-v opened the pull request:
**Component Created:** BlockButtons.vue

**Stories:** ~/stories/BlockButtons.stories.js

**Spec:** ~/stories/BlockButtons.spec.js

**Notes:**

This component is just a wrapper for buttons so we can align them as we want


I left a note in a `ButtonLink` component regarding its prop definition. Since we are using TypeScript we want to use its functionalities. 

I also implemented `ButtonLinkRefactored` which is using the props in a TypeScript way, and it can be used as an example of how to define component enums + props and use them across the app.


<img width="906" height="586" alt="image" src="https://github.com/user-attachments/assets/e9513f84-c64d-4ee0-aad7-6164fb496d23" />



**Component screenshot:**

<img width="1045" height="898" alt="image" src="https://github.com/user-attachments/assets/ef24c931-c9ec-4b4f-8326-8b55e7200729" />


**Checklist:**

-   [ ] I checked that it is working locally in the dev server
-   [x] I checked that it is working locally in the storybook
-   [ ] I checked that it is working locally in the 
library-website-nuxt dev server
-   [x] I added a screenshot of it working
-   [ ] UX has reviewed and approved this
-   [ ] I assigned this PR to someone on the dev team to review
-   [x] I used a conventional commit message
-   [x] I assigned myself to this PR

pghorpade commented:
@lux-v I am a bit confused why we need the BlockButton component. Can you post the Figma design where this will be used on the page template? If it is only used on the page template, can we simply add the styles to the page?

Additionally, at this time, we don't want to refactor ButtonLink, as it is used on many sites. And we won't be able to update it right now if we merge the refactored button. Also, primary, secondary, and tertiary are more for adding different colors like text and background to the button, but download is not adding any special colors, so it's different from the rest of the props.

lux-v commented:
> @lux-v I am a bit confused why we need the BlockButton component. Can you post the Figma design where this will be used on the page template? If it is only used on the page template, can we simply add the styles to the page?
> 
> Additionally, at this time, we don't want to refactor ButtonLink, as it is used on many sites. And we won't be able to update it right now if we merge the refactored button. Also, primary, secondary, and tertiary are more for adding different colors like text and background to the button, but download is not adding any special colors, so it's different from the rest of the props.

@pghorpade `BlockButton` is used here in [this](https://www.figma.com/design/CDhWDARLb36ftkQce1LyLC/Breakpoints?node-id=2023-6669&m=dev) section, and in any section where multiple buttons might be needed.

The goal is to have a component that ensures a consistent design and allows flexible use of multiple buttons anywhere on the site. For example, another button might be added in [this](https://www.figma.com/design/CDhWDARLb36ftkQce1LyLC/Breakpoints?node-id=2076-7931&m=dev) section in the future.  


I also added notes regarding the refactored button: I removed the download variant and made it a separate prop. 

I created a new `ButtonLinkRefactored` component to provide an example for you guys of how to use it with TypeScript, and left the existing ButtonLink component untouched. Let me know if that's ok.

pghorpade commented:
@lux-v there are eslint errors on this PR

pghorpade commented:
@lux-v I looked at your comment and saw the design where this component could be used. I am fine with keeping this component based on how it is used and could be used in future.

pghorpade reviewed (approved):
👍 Thank you for the changes

pghorpade commented:
:tada: This PR is included in version @ucla-library-monorepo/ucla-library-website-components-v1.33.0 :tada:

The release is available on [GitHub release](https://github.com/UCLALibrary/ucla-library-website-components/releases/tag/%40ucla-library-monorepo/ucla-library-website-components-v1.33.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
